staging/src/k8s.io/kms/encryption/aesgcm.go

```diff
@@ -0,0 +1,100 @@
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"errors"
+)
+
+var (
+	ErrEmptyPlaintext = errors.New("plaintext for encryption is empty")
+)
+
+// AESGCM is a struct that contains a key of random bytes and additional
+// information to re-use it in a safe way.
+type AESGCM struct {
+	key []byte
+
+	aesgcm cipher.AEAD
+}
+
+// NewAESGCM returns a pointer to a AESGCM with a key and cipher.AEAD created.
+func NewAESGCM() (*AESGCM, error) {
+	key, err := randomBytes(keySize)
+	if err != nil {
+		return nil, err
+	}
+
+	return FromKey(key)
+}
+
+func FromKey(key []byte) (*AESGCM, error) {
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+
+	return &AESGCM{
+		key:    key,
+		aesgcm: aesgcm,
+	}, nil
+
+}
+
+// Key returns the internal Key used for encryption. One should be cautious with
+// this value as it is confidential data that must not leak.
+func (c *AESGCM) Key() []byte {
+	return c.key
+}
+
+// Encrypt encrypts given plaintext. The nonce is prepended. Therefore any
```

Comment:
Could you add more details in the comment about why a nonce is prepended?

Reply:
Using `aes.NewGCMTransformer`, no reason to mention it any more 😄